src/tokenize.cpp

```diff
@@ -72,28 +72,28 @@ struct tok_ctx
 
    bool more()
    {
-      return(c.idx < (int)data.size());
+      return(c.idx < data.size());
    }
 
 
-   int peek()
+   size_t peek()
    {
-      return(more() ? data[c.idx] : -1);
+      return(more() ? data[c.idx] : 0);
    }
 
 
-   int peek(int idx)
+   size_t peek(size_t idx)
    {
       idx += c.idx;
-      return((idx < (int)data.size()) ? data[idx] : -1);
+      return((idx < data.size()) ? data[idx] : 0);
    }
 
 
-   int get()
+   size_t get()
```

Comment:
This is correct.
Even if the tests are OK and the letters, punctuators and digits are always positives
I would like to discuss what could be the better solution.